### DISCUSSION TITLE
Add server-side and client-side entry points to App router

### DISCUSSION THREAD

**Discussion by olivierr91** [vercel/next.js#63732] [Ideas]
### Goals

1. Initialize server-side librairies
2. Initialize client-side librairies
3.

### Non-Goals

1.
2.
3.

### Background

Next feels like is missing a big hook in application lifecycle. There is currently no way to initialize things, except calling initialization routines at the top of every page and every layouts. It feels like a design issue for any serious application.

I am aware of instrumentation.ts, but it runs in a separate Node execution context and is run only once, so everything initialized in there is not accessible in other parts of the app (like pages and layouts). Initializing DI containers such as tsyringe in instrumentation.ts and accessing them in other parts of the app does not work.

### Proposal

Add an server-side and client-side entry-point that is run on every request or app initialization.

**Comment by moneydance** [comment]
Running into this now is there a workaround?
